scheduler/src/helper/generator.helper.ts

```diff
@@ -0,0 +1,443 @@
+ * 1. Cron Job Scheduling
+ *    - A cron job is configured to run at 5:30 PM on workdays (Monday to Friday).
+ *    - The cron job initiates the CGI file generation and uploading process.
+ * 
+ * 2. Transaction Data Retrieval
+ *    - The CGI file generator script is executed by the cron job.
+ *    - The script queries the [ORBC_CFS_TRANSACTION_DETAIL] table in the database to retrieve transactions with a status marked as "READY".
+ * 
+ * 3. CGI File Generation
+ *    - For each "READY" transaction, the CGI file generator:
+ *      - Populates the CGI file with the necessary transaction data (batch header, journal header and journal details) according to the predefined CGI file layout.
+ *      - Creates a corresponding CGI trigger file to indicate that the transfer of INBOX file(s) is complete.
+ * 
+ * 4. Temporary Storage
+ *    - The generated CGI files and their associated trigger files are temporarily stored in a designated local directory.
+ * 
+ * 5. Uploading to SFTP Server
+ *    - The CGI file generator script establishes a connection to the SFTP server.
+ *    - The script uploads the CGI files and trigger files to the appropriate directory on the SFTP server.
+ * 
+ * 6. Completion
+ *    - Once all files are uploaded, the script terminates.
+ *    - The cron job completes its execution for the day.
+ * 
+ * Future Enhancements
+ * 
+ * The current workflow does not include file verification, error handling, or notifications. These features can be added in future iterations to improve reliability and transparency:
+ * 
+ * - File Verification
+ *   - Implement checks to ensure that generated CGI files match expected formats and data integrity is maintained before uploading.
+ * 
+ * - Error Handling
+ *   - Incorporate mechanisms to handle errors during file generation and upload processes.
+ *   - Implement retry logic for transient errors.
+ * 
+ * - Notifications
+ *   - Set up notifications to alert relevant personnel in case of errors or successful completions.
+ */
+
+import * as fs from 'fs';
+import { join} from 'path';
+import { CgiSftpService } from "src/modules/cgi-sftp/cgi-sftp.service";
+import { Transaction } from "src/modules/transactions/transaction.entity";
+import { Readable } from "typeorm/platform/PlatformTools";
+import { CgiConstants } from "src/common/constants/cgi.constant";
+import { randomBytes } from 'crypto';
+
+const maxBatchId = '';
+
+export class BatchHeader {
+  feederNumber: string;
+  batchType: string;
+  transactionType: string;
+  fiscalYear: number;
+  batchNumber: string;
+  messageVersion: string;
+
+  constructor() {
+    this.feederNumber = process.env.FEEDER_NUMBER;
+    this.batchType = CgiConstants.BATCH_TYPE;
+    this.transactionType = CgiConstants.TRANSACTION_TYPE_BH;
+    this.fiscalYear = getFiscalYear();
+    this.batchNumber = getBatchNumber();
+    this.messageVersion = CgiConstants.MESSAGE_VERSION;
+  } 
+}
+
+  export function formatDateToCustomString(date: Date): string {
```

Comment:
Can this be formatted with the help of dayjs library function?
```
dayjs(date).format('YYYYMMDDHHmmss')
```
Doesn't this give the same output?